### PR TITLE
MOROCO

### PR DESCRIPTION
**Pull request by MihaelaGaman**
Add MOROCO to huggingface datasets.

### CODE REVIEW

File: datasets/moroco/moroco.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            - tech
+"""
+
+_HOMEPAGE = "https://github.com/butnaruandrei/MOROCO"
+
+_LICENSE = "CC BY-SA 4.0 License"
+
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URL = "https://raw.githubusercontent.com/MihaelaGaman/MOROCO/huggingface-datasets-addition/MOROCO/preprocessed/all/"
```
Comment (lhoestq) [inline datasets/moroco/moroco.py:51]: Why not use the link from the official repo ?
You can have something like
```python
_URL = "https://github.com/butnaruandrei/MOROCO/tree/master/MOROCO/preprocessed/train/"

_TRAIN_SAMPLES_FILE = "train/samples.txt"
_TRAIN_LABELS_FILE = "train/category_labels.txt"

_VAL_SAMPLES_FILE = "validation/samples.txt"
_VAL_LABELS_FILE = "validation/category_labels.txt"

_TEST_SAMPLES_FILE = "test/samples.txt"
_TEST_LABELS_FILE = "test/category_labels.txt"
```

File: datasets/moroco/README.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
+---
+annotations_creators:
+- found
+language_creators:
+- found
+languages:
+- ro, md
```
Comment (lhoestq) [inline datasets/moroco/README.md:7]:
```suggestion
- ro-md
```
We're using the BCP-47 languages codes that includes the ISO-639-1 codes such as `ro-md` for Romanian (Republic of Moldova).

If you think that's more relevant we can also use
```suggestion
- ro
- ro-md
```

File: datasets/moroco/README.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+licenses:
+- cc-by-4.0
+multilinguality:
+- monolingual
+size_categories:
+- n>30K
```
Comment (lhoestq) [inline datasets/moroco/README.md:13]: We only use the buckets "n<1K", "1K<n <10K", "10K< n<100K", "100K<n<1M" and "n>1M".

So you can use instead:
```suggestion
- 10K<n<100K
```

File: datasets/moroco/README.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+  - [Citation Information](#citation-information)
+  - [Contributions](#contributions)
+
+## Dataset Description
+
+- **Homepage:** https://github.com/butnaruandrei/MOROCO
+- **Repository:** https://github.com/butnaruandrei/MOROCO
+- **Paper:** Andrei M. Butnaru, Radu Tudor Ionescu. MOROCO: The Moldavian and Romanian Dialectal Corpus. ArXiv, 2019 [(link to paper)](https://arxiv.org/abs/1901.06543).
+- **Leaderboard:** [Needs More Information]
+- **Point of Contact:** raducu.ionescu@gmail.com
```
Comment (lhoestq) [inline datasets/moroco/README.md:54]:
```suggestion
- **Homepage:** [Github](https://github.com/butnaruandrei/MOROCO)
- **Repository:** [Github](https://github.com/butnaruandrei/MOROCO)
- **Paper:** [Arxiv](https://arxiv.org/abs/1901.06543)
- **Leaderboard:** [Needs More Information]
- **Point of Contact:** raducu.ionescu@gmail.com
```

File: datasets/moroco/README.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+### Citation Information
+
+@inproceedings{ Butnaru-ACL-2019,
+    author = {Andrei M. Butnaru and Radu Tudor Ionescu},
+    title = "{MOROCO: The Moldavian and Romanian Dialectal Corpus}",
+    booktitle = {Proceedings of ACL},
+    year = {2019},
+    pages={688--698},
+}
```
Comment (lhoestq) [inline datasets/moroco/README.md:164]:
````suggestion
```
@inproceedings{ Butnaru-ACL-2019,
    author = {Andrei M. Butnaru and Radu Tudor Ionescu},
    title = "{MOROCO: The Moldavian and Romanian Dialectal Corpus}",
    booktitle = {Proceedings of ACL},
    year = {2019},
    pages={688--698},
}
```
````
add code block

File: datasets/moroco/moroco.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+_DESCRIPTION = """\
+        The MOROCO (Moldavian and Romanian Dialectal Corpus) dataset contains 33564 samples of text collected from the news domain.
+        The samples belong to one of the following six topics:
+            - culture
+            - finance
+            - politics
+            - science
+            - sports
+            - tech
+"""
```
Comment (lhoestq) [inline datasets/moroco/moroco.py:43]:
```suggestion
_DESCRIPTION = """\
The MOROCO (Moldavian and Romanian Dialectal Corpus) dataset contains 33564 samples of text collected from the news domain.
The samples belong to one of the following six topics:
    - culture
    - finance
    - politics
    - science
    - sports
    - tech
"""
```

**Review by lhoestq — COMMENTED** [review]
Nice ! Thanks for adding this one

I just left a few comments

File: datasets/moroco/moroco.py (hunk 1 above)
Comment (MihaelaGaman) [reply datasets/moroco/moroco.py:51]: I just got a better data format in the official repo, so I'll go ahead and just use that link in an upcoming commit.

Comment (lhoestq) [reply datasets/moroco/moroco.py:51]: Ok thanks !

**Comment by MihaelaGaman** [comment]
@lhoestq Thank you for all the feedback. I've added the suggested changes in my last commit.

**Review by lhoestq — APPROVED** [review]
Looks all good !
Thanks :)

**Pull request merged**